src/plots/cartesian/axes.js

```diff
@@ -1203,6 +1210,53 @@ axes.calcTicks = function calcTicks(ax, opts) {
     return ticksOut;
 };
 
+function filterRangeBreaks(ax, ticksOut) {
+    if(ax.rangebreaks) {
+        // remove ticks falling inside rangebreaks
+        ticksOut = ticksOut.filter(function(d) {
+            return ax.maskBreaks(d.x) !== BADNUM;
+        });
+    }
+
+    return ticksOut;
+}
+
+function syncTicks(ax) {
+    // get the overlaying axis
+    var baseAxis = ax._mainAxis;
+
+    var ticksOut = [];
+    if(baseAxis._vals) {
+        for(var i = 0; i < baseAxis._vals.length; i++) {
+            // get the position of the every tick
+            var pos = baseAxis.l2p(baseAxis._vals[i].x);
+
+            // get the tick for the current axis based on position
+            var vali = ax.p2l(pos);
+            var val1 = ax.p2l(pos - 0.2);
+            var val2 = ax.p2l(pos + 0.2);
+            var d = 1 + Math.round(Math.log10(Math.abs(val2 - val1)));
+            var e = Math.pow(10, -d);
+            var valR = Math.round(vali * e) / e;
+            var objR = axes.tickText(ax, valR);
+            var obj = axes.tickText(ax, vali);
+            obj.text = objR.text;
+
```

Comment:
Now to be more accurate about the position of labels and ticks lets simplify this part 
from:
```js
            var objR = axes.tickText(ax, valR);
            var obj = axes.tickText(ax, vali);
            obj.text = objR.text;
```
to:
```js
            var obj = axes.tickText(ax, valR);
```

Reply:
With this change, the scroll behaviour is not perfect. It flicks when we try to scroll. Should I commit the changes?

Reply:
Yes please commit the change. One may argue that having a more accurate static graph is more important.
After that to reduce the flicks we may end up reducing the deltas from 0.2 to 0.15 or even 0.1 here:
```js
            var val1 = ax.p2l(pos - 0.1);
            var val2 = ax.p2l(pos + 0.1);
```

Reply:
What exactly do you mean by "it flicks"? Yes, the static graph needs to be accurate, but can't we get that and clean scroll behavior simultaneously?

Reply:
> 
> ```js
>             var val1 = ax.p2l(pos - 0.1);
>             var val2 = ax.p2l(pos + 0.1);
> ```

@alexcjohnson This fixes the "flicks".
 @archmoj Clean and smooth now.